File: spring-data/src/main/java/spring/playground/springdata/persistence/repository/MemberJpaRepository.java
package spring.playground.springdata.persistence.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import spring.playground.springdata.persistence.entity.member.Member;

import java.util.List;

public interface MemberJpaRepository extends JpaRepository<Member, Long> {
    List<Member> findByName(String name);
}

File: spring-kotlin-user/src/main/kotlin/com/example/kotlin/user/model/repository/MemberJpaRepository.kt
package com.example.kotlin.user.model.repository

import com.example.kotlin.user.model.entity.MemberEntity
import org.springframework.data.jpa.repository.JpaRepository

interface MemberJpaRepository: JpaRepository<MemberEntity, Long>{
    fun findByLoginId(loginId: String): MemberEntity?
    fun findByEmail(email: String): MemberEntity?
    fun findByLoginIdOrEmail(loginId: String, email: String): MemberEntity?
}